duecredit/collector.py

```diff
@@ -454,7 +479,11 @@ def __init__(self, collector, outputs="stdout,pickle", fn=DUECREDIT_FILE):
         ]
 
     @staticmethod
-    def _get_output_handler(type_, collector, fn=None):
+    def _get_output_handler(
```

Comment:
I suggest annotating `type_` as `Literal["stdout", "stderr", "pickle"]` and using [`@typing.overload`](https://docs.python.org/3/library/typing.html#typing.overload) to select the correct return type.

Reply:
Or an Enum, I don't know if this is a public API